src/Streamly/Extra/Metrics.hs

```diff
@@ -124,8 +124,10 @@ type MaybeUpdateFn = Maybe (Double -> Double)
 
 data MetricDetails =
   MetricDetails
-    { counters :: [(Counter, MaybeUpdateFn)]
-    , gauges   :: [(Gauge, MaybeUpdateFn)]
+    { counters        :: [(Counter, MaybeUpdateFn)]
+    , gauges          :: [(Gauge, MaybeUpdateFn)]
+    , vector1Counters :: [(Vector P.Label1 Counter, MaybeUpdateFn)]
+    , vector1Gauges   :: [(Vector P.Label1 Gauge, MaybeUpdateFn)]
```

Comment:
should we pass the value of label here itself, rather than using it as `LoggerDetails.tag`?

Reply:
Yeah, that makes sense. Does a `([Vector Label s, MaybeUpdateFn], Text)` be a good structure? `s` being `Counter/Gauge` in this case.